### PR TITLE
zephyr: cas: Added support for testing Common Audio Service

### PR DESCRIPTION
**Pull request by MSEEHenrik**
- Set Member Lock
- Get Member RSI

### CODE REVIEW

**Review by mkasenberg — CHANGES_REQUESTED** [review]

File: autopts/ptsprojects/zephyr/cap.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    stack = get_stack()
+
+    advData, rspData = {}, {}
+
+    pre_conditions = [
+        TestFunc(btp.core_reg_svc_gap),
+        TestFunc(stack.gap_init, iut_device_name),
+        TestFunc(btp.gap_read_ctrl_info),
+        TestFunc(lambda: pts.update_pixit_param("CAP", "TSPX_iut_use_dynamic_bd_addr",
+            "TRUE" if stack.gap.iut_addr_is_random() else "FALSE")),
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/ptsprojects/zephyr/cap.py]: Strange, I didn't get any exception, but I removed the parameter update.

File: autopts/ptsprojects/zephyr/cas.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    stack = get_stack()
+
+    pre_conditions = [
+        TestFunc(btp.core_reg_svc_gap),
+        TestFunc(stack.gap_init, iut_device_name),
+        TestFunc(btp.gap_read_ctrl_info),
+        TestFunc(lambda: pts.update_pixit_param("CAS", "TSPX_bd_addr_iut",
+            stack.gap.iut_addr_get_str())),
+        TestFunc(lambda: pts.update_pixit_param("CAS", "TSPX_iut_use_dynamic_bd_addr",
+            "TRUE" if stack.gap.iut_addr_is_random() else "FALSE")),
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/ptsprojects/zephyr/cas.py]: Strange, I didn't get any exception, but I removed the parameter update.

File: autopts/wid/cas.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return generic_wid_hdl(wid, description, test_case_name, [__name__])
+
+
+# wid handlers section begin
+
+"""
+    Please prepare IUT into a connectable mode.
+    Description: Verify that the Implementation Under Test (IUT) can accept GATT connect request from PTS.
+"""
+def hdl_wid_20001(_:WIDParams):
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/wid/cas.py]: Done.

File: autopts/wid/cas.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+import logging
+import re
+import sys
+
+from enum import IntFlag
+from autopts.pybtp import btp, defs
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/wid/cas.py]: Done.

File: autopts/wid/cap.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# This program is distributed in the hope it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+import logging
+import re
+import sys
+from enum import IntEnum
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/wid/cap.py]: Done.

File: autopts/wid/cas.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+def hdl_wid_20001(_:WIDParams):
+    stack = get_stack()
+    btp.gap_set_conn()
+    btp.gap_adv_ind_on(ad=stack.gap.ad)
+    return True
+
+"""
+    Please remove LTK from IUT before proceeding.
+"""
+def hdl_wid_20128(_:WIDParams):
```
[the remark this answers is not in the document]
Comment (MSEEHenrik) [reply autopts/wid/cas.py]: Done.

File: autopts/ptsprojects/zephyr/cas.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkasenberg) [reply autopts/ptsprojects/zephyr/cas.py]: It appears only in autoptsserver logs and does not interrupt running test cases.

**Review by mkasenberg — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**